### PR TITLE
Modify ItOperatorWlsUpgrade for v9 cluster reference

### PR DESCRIPTION
**Pull request by maggiehe00**
This PR is to use checkin in domain yaml file to test Opeartor WLS upgrade from 3.1.4, 3.2.5, 3.3.8, 3.4.0, and 3.4.1 to the latest WKO 4.0 v9 cluster reference and verify correspond cluster scaling after operator gets upgrade.
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12680/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12681/

Right now except 3.1.4 to WKO 4.0 failed all the other versions work fine for domain in image, mii and mii Aux. Temporarily disables this failing method and logged a track JIRA [OWLS-102316](https://jira.oraclecorp.com/jira/browse/OWLS-102316)

The following Kind tests is after disabling testOperatorWlsUpgradeFrom314ToCurrent (still running)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12745/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/12746/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 1]
```diff
@@ -658,4 +656,82 @@ private void checkAdminPortForwarding(String domainNamespace, boolean successExp
     stopPortForwardProcess(domainNamespace);
   }
 
+  private void createWlsDomainAndVerifyByDomainYaml(String domainType,
```
[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:659]: Yes ,  After operator is upgraded, I check if the (new) Cluster Resource is created and we can scale the cluster using cluster resource
Line 417 upgradeOperatorToCurrent(opNamespace,domainNamespace,domainUid);
Line 434 scaleClusterUpAndDown();

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:667]: Description is added to createWlsDomainAndVerifyByDomainYaml

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 2]
```diff
@@ -158,7 +165,10 @@ public static void init() {
 
   /**
    * Operator upgrade from 3.1.4 to current.
+   * Currently form 3.1.4 to WKO 4.0, v9 cluster reference doesn't work
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:168]: typo

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 3]
```diff
@@ -322,16 +331,18 @@ private void scaleClusterUpAndDown() {
...
+            clusterName, domainNamespace));
 
-    scaleAndVerifyCluster("cluster-1", domainUid, domainNamespace,
-        managedServerPodNamePrefix, replicaCount, 2,
-        true, externalRestHttpsPort, opNamespace, opServiceAccount,
-        false, "", "", 0, "", "", null, null);
+    logger.info("Updating the cluster {0} replica count to 2");
+    p1Success = scaleCluster(clusterName, domainNamespace,2);
+    assertTrue(p1Success,
+        String.format("Patching replica to 1 failed for cluster %s in namespace %s",
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:344]: incorrect replica count in error message

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 4]
```diff
@@ -658,4 +656,90 @@ private void checkAdminPortForwarding(String domainNamespace, boolean successExp
     stopPortForwardProcess(domainNamespace);
   }
 
+  /**
+   * Replace the fields in the pre-checked in domain yaml file with testing attributes.
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:660]:
```suggestion
   * Replace the fields in domain yaml file with testing attributes.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 5]
```diff
@@ -658,4 +656,90 @@ private void checkAdminPortForwarding(String domainNamespace, boolean successExp
...
+   * For example, namespace, domainUid,  and image. Then create domain using kubectl and
+   * verify the created domain
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:662]:
```suggestion
   * verify the domain is created.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java [hunk 6]
```diff
@@ -658,4 +656,90 @@ private void checkAdminPortForwarding(String domainNamespace, boolean successExp
...
+   * @param domainType either domain in image(Image) or model in image (FromModel)
+   * @param domainNamespace namespace where domain exists
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java:664]:
```suggestion
   * @param domainNamespace namespace in which to create domain resource
```

**Review by sankarpn — CHANGES_REQUESTED** [review]

**Review by anpanigr — APPROVED** [review]
lgtm

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java (hunk 6 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java (hunk 5 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java (hunk 4 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java (hunk 3 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java (hunk 2 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorWlsUpgrade.java]: Corrected

**Review by sankarpn — APPROVED** [review]
lgtm

**Pull request merged**